src/NuGet.Clients/PackageManagement.UI/PackageLoader.cs

```diff
@@ -509,10 +533,27 @@ private IEnumerable<string> GetSupportedFrameworks()
                 cancellationToken.ThrowIfCancellationRequested();
                 resultCount++;
 
-                var searchResultPackage = new SearchResultPackageMetadata(_sourceRepository);
+                var searchResultPackage = new SearchResultPackageMetadata();
                 searchResultPackage.Id = package.Identity.Id;
                 searchResultPackage.Version = package.Identity.Version;
+                searchResultPackage.IsSolution = _isSolution;
+                searchResultPackage.IsUpdateTab = _option.Filter == Filter.UpdatesAvailable;
```

Comment:
I would put the logic on what to show in the view, and not in the viewmodel of an individual search result